ntp-metrics-exporter/src/lib.rs

```diff
@@ -78,50 +52,33 @@ pub async fn main() -> Result<(), Box<dyn std::error::Error>> {
         },
     };
 
-    let app = Router::new()
-        .route(
-            "/metrics",
-            get(|| async {
-                let mut stream = tokio::net::UnixStream::connect(observation_socket_path).await?;
-                let mut msg = Vec::with_capacity(16 * 1024);
-                let output: ObservableState =
-                    ntp_daemon::sockets::read_json(&mut stream, &mut msg).await?;
-                let metrics = Metrics::default();
-                metrics.fill(&output);
-                let registry = metrics.registry();
-                let mut buf = String::new();
-                prometheus_client::encoding::text::encode(&mut buf, &registry)?;
-                Ok::<_, ServeError>((
-                    HeaderMap::from_iter([(
-                        axum::http::header::CONTENT_TYPE,
-                        HeaderValue::from_static("text/plain"),
-                    )]),
-                    buf,
-                ))
-            }),
-        )
-        .route(
-            "/",
-            get(|| async {
-                (
-                    StatusCode::FOUND,
-                    AppendHeaders([(axum::http::header::LOCATION, "/metrics")]),
-                )
-            }),
-        )
-        .fallback(|| async {
-            (
-                StatusCode::NOT_FOUND,
-                Json(json!({
-                    "error": 404,
-                })),
-            )
-        });
-
-    axum::Server::bind(&cli.listen_socket)
-        .serve(app.into_make_service())
-        .await
-        .unwrap();
-
-    Ok(())
+    println!("starting ntp-metrics-exporter on {}", &cli.listen_socket);
+
+    let listener = TcpListener::bind(cli.listen_socket).await?;
+
+    loop {
+        let (mut tcp_stream, _) = listener.accept().await?;
+
+        let mut stream = tokio::net::UnixStream::connect(&observation_socket_path).await?;
+        let mut msg = Vec::with_capacity(16 * 1024);
+        let output: ObservableState = ntp_daemon::sockets::read_json(&mut stream, &mut msg).await?;
+        let metrics = Metrics::default();
+        metrics.fill(&output);
+        let registry = metrics.registry();
+
+        let mut content = String::with_capacity(4 * 1024);
+        prometheus_client::encoding::text::encode(&mut content, &registry)?;
+
+        let mut buf = String::with_capacity(4 * 1024);
+
+        // headers
+        buf.push_str("HTTP/1.1 200 OK\n");
+        buf.push_str("content-type: text/plain\n");
+        write!(buf, "content-length: {}\n\n", content.len()).unwrap();
```

Comment:
Per the http spec (https://www.ietf.org/rfc/rfc2616.txt page 15) these line feeds should be preceeded by carriage returns.